coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentControllerBase.java

```diff
@@ -394,11 +395,21 @@ protected ModelAndView narrativePageSave(ProposalDevelopmentDocumentForm form, b
         ProposalDevelopmentDocument document = (ProposalDevelopmentDocument) getDocumentService().getByDocumentHeaderId(form.getDocument().getDocumentNumber());
         if (canEdit) {
             if (new ProposalDevelopmentDocumentRule().processAttachmentRules(form.getProposalDevelopmentDocument())) {
-                //when saving on a page in the narrative locking region we don't want to over write proposal locking region data,
-                //so we retrieve the latest proposal from the db, and replace the current propopsal with that, and then copy the attachments
-                document.getDevelopmentProposal().setNarratives(form.getDevelopmentProposal().getNarratives());
-                document.getDevelopmentProposal().setInstituteAttachments(form.getDevelopmentProposal().getInstituteAttachments());
-                document.getDevelopmentProposal().setPropPersonBios(form.getDevelopmentProposal().getPropPersonBios());
+                for(Narrative narrative : form.getDevelopmentProposal().getNarratives() ) {
+                	if(narrative.getObjectId() == null || narrative.isUpdated()) {
+                		document.getDevelopmentProposal().getNarratives().add(narrative);
```

Comment:
Still a little confused about this. So what happens in the case where the narrative existed, but was updated? Won't it end up in this list twice? And that doesn't cause problems?